notifications_utils/letter_timings.py

```diff
@@ -84,9 +87,13 @@ def letter_can_be_cancelled(notification_status, notification_created_at):
     if notification_status not in ('created', 'pending-virus-check'):
         return False
 
-    if _after_letter_processing_deadline() and _notification_created_before_deadline(notification_created_at):
+    if _after_letter_processing_deadline() and _notification_created_before_today_deadline(notification_created_at):
         return False
 
+    if _notification_created_before_that_day_deadline(
+        notification_created_at
+    ) and notification_created_at.date() < datetime.utcnow().date():
```

Comment:
How can the `notification_created_at` be less than `now`?

Reply:
it's always less than now, past dates are smaller than present dates, no? So when date is yesterday it's smaller than today's date.

Reply:
Instead can we just do something like `notification_created_at < today + deadline time`?
It means that as long as the status is in allowable and the notification is before today's deadline then we can cancel the notification.

Reply:
Ok - I've finally wrapped my head around all the possibilities. This is probably for the best because we are only in a situation of having a letter that is not sent and from yesterday if the send to provider task failed. Since that task would likely be started at some point during the day, it's difficult to allow a letter to be cancelled while it is being sent. So it's best if we decline the cancellation.